dokka-subprojects/plugin-base-frontend/src/main/ui-kit/code-block/index.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2014-2025 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+import './styles.scss';
+
+// helps with some corner cases where <wbr> starts working already,
+// but the signature is not yet long enough to be wrapped
+
+const CODE_BLOCK_PADDING = 16 * 2;
+
+const symbolsObserver = new ResizeObserver((entries) => entries.forEach(wrapSymbolParameters));
+
+function initHandlers() {
+  document.querySelectorAll('div.symbol').forEach((symbol) => symbolsObserver.observe(symbol));
+}
+
+if (document.readyState === 'loading') {
+  window.addEventListener('DOMContentLoaded', initHandlers);
+} else {
+  initHandlers();
+}
+
+function createNbspIndent() {
+  const indent = document.createElement('span');
+  indent.append(document.createTextNode('\u00A0\u00A0\u00A0\u00A0'));
+  indent.classList.add('nbsp-indent');
+  return indent;
+}
+
+function wrapSymbolParameters(entry: ResizeObserverEntry) {
```

Comment:
we could wrap the body of this function into requestAnimationFrame call in order to avoid `ResizeObserver loop completed with undelivered notifications` kind of errors that are likely caused by using getBoundingClientRect

Reply:
Added